standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/columnstats/aggr/LongColumnStatsAggregatorTest.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.hadoop.hive.metastore.columnstats.aggr;
+
+import org.apache.hadoop.hive.common.ndv.fm.FMSketch;
+import org.apache.hadoop.hive.common.ndv.hll.HyperLogLog;
+import org.apache.hadoop.hive.metastore.StatisticsTestUtils;
+import org.apache.hadoop.hive.metastore.TableType;
+import org.apache.hadoop.hive.metastore.annotation.MetastoreUnitTest;
+import org.apache.hadoop.hive.metastore.api.ColumnStatistics;
+import org.apache.hadoop.hive.metastore.api.ColumnStatisticsData;
+import org.apache.hadoop.hive.metastore.api.ColumnStatisticsObj;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.metastore.utils.MetaStoreServerUtils.ColStatsObjWithSourceInfo;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.apache.hadoop.hive.metastore.StatisticsTestUtils.assertLongStats;
+import static org.apache.hadoop.hive.metastore.StatisticsTestUtils.createFMSketch;
+import static org.apache.hadoop.hive.metastore.StatisticsTestUtils.createHll;
+import static org.apache.hadoop.hive.metastore.Warehouse.DEFAULT_CATALOG_NAME;
+
+@Category(MetastoreUnitTest.class)
+public class LongColumnStatsAggregatorTest {
+
+  private static final Table TABLE = new Table("dummy", "db", "hive", 0, 0,
+      0, null, null, Collections.emptyMap(), null, null,
+      TableType.MANAGED_TABLE.toString());
+  private static final FieldSchema COL = new FieldSchema("col", "int", "");
+
+  @Test
+  public void testAggregateSingleStat() throws MetaException {
+    LongColumnStatsAggregator aggregator = new LongColumnStatsAggregator();
+
+    List<String> partitionNames = Collections.singletonList("part1");
+
+    HyperLogLog hll = createHll(1, 3);
+    ColumnStatisticsData data1 = StatisticsTestUtils.createLongStats(
```

Comment:
In Hive, the Line length limit is 120 chars, instead of 80 chars ([coding conventions](https://cwiki.apache.org/confluence/display/Hive/HowToContribute#HowToContribute-CodingConventions)) so you don't need to add the line break unless you want to.

Reply:
The hard limit of 120 chars does not really take into account rendering in modern IDE where the parameter names are often displayed, when I find it convenient I prefer to split long lines to improve readability (which I understand is highly subjective and linked to my specific setup in terms of font size and resolution which is non-standard).

If that's OK with you, I would leave the line breaks as-is.

Reply:
Sure, I just though was worth mentioning in case you were not aware off.

Off-topic, but I kind of get annoyed some times when the IDE renders the parameters names so I disable them :D